### PR TITLE
Feat: Dynamic kubernetes api for classless api requesting

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

File: util/src/main/java/io/kubernetes/client/util/generic/dynamic/DynamicKubernetesApi.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        DynamicKubernetesListObject.class,
+        apiGroup,
+        apiVersion,
+        resourcePlural,
+        apiClient);
+    TypeAdapter<KubernetesObject> objAdapter =
+        apiClient.getJSON().getGson().getAdapter(KubernetesObject.class);
+    TypeAdapter<KubernetesListObject> objListAdapter =
+        apiClient.getJSON().getGson().getAdapter(KubernetesListObject.class);
+    if (false
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/generic/dynamic/DynamicKubernetesApi.java:52]: delete `false`? It's not doing anything here.

**Comment by brendandburns** [comment]
One comment on some stray code. I think this would also benefit from an example to show how to use it.

File: util/src/main/java/io/kubernetes/client/util/generic/dynamic/DynamicKubernetesApi.java (hunk 1 above)
Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/util/generic/dynamic/DynamicKubernetesApi.java]: deleted

**Comment by brendandburns** [comment]
@yue9944882 Is it possible to add an example? If not, I can lgtm and we can add it later.

**Comment by yue9944882** [comment]
@brendandburns just added an example for the dynamic client as well as the wiki page

**Comment by davidxia** [comment]
Thanks so much for this feature! I love it, and my team is using it now.

**Pull request merged**